src/vkvg_pattern.c

```diff
@@ -172,6 +172,8 @@ vkvg_status_t vkvg_pattern_get_color_stop_rgba (VkvgPattern pat, uint32_t index,
 												float* offset, float* r, float* g, float* b, float* a) {
 	if (pat->type == VKVG_PATTERN_TYPE_SURFACE || pat->type == VKVG_PATTERN_TYPE_SOLID)
 		return VKVG_STATUS_PATTERN_TYPE_MISMATCH;
+	if (index >= 16)
+		return VKVG_STATUS_INVALID_INDEX;
 	vkvg_gradient_t* grad = (vkvg_gradient_t*)pat->data;
```

Comment:
gradient has a count field, invalid index must be returned only if index >= grad->count

Reply:
Done